src/Cody.VisualStudio.Completions/Completions/CodyProposalManager.cs

```diff
@@ -6,32 +8,50 @@ namespace Cody.VisualStudio.Completions
 {
     public class CodyProposalManager : ProposalManagerBase
     {
+        private readonly ILog _logger;
         private static TraceLogger trace = new TraceLogger(nameof(CodyProposalManager));
 
+        public CodyProposalManager(ILog logger)
+        {
+            _logger = logger;
+        }
+
         public override bool TryGetIsProposalPosition(VirtualSnapshotPoint caret, ProposalScenario scenario, char triggerCharacter, ref bool value)
         {
-            switch (scenario)
+            try
+            {
+                switch (scenario)
+                {
+                    case ProposalScenario.TypeChar:
+                        trace.TraceEvent("TypeCharScenario");
+                        if (char.IsWhiteSpace(triggerCharacter) && caret.Position.Position >= 2)
+                        {
+                            char c = caret.Position.Snapshot[caret.Position.Position - 2];
+                            if (!char.IsWhiteSpace(c)) value = true;
+                        }
+                        else value = true;
+
+                        break;
+                    case ProposalScenario.Return:
+                        trace.TraceEvent("ReturnScenario");
+                        if (caret.Position.GetContainingLine().End == caret.Position) value = true;
+                        break;
+                    default:
+                        trace.TraceEvent("OtherScenario");
+                        value = true;
+                        break;
+                }
+
+
+                trace.TraceEvent("ShowProposal", value);
+                return value;
```

Comment:
I think you can simplify that by removing that `value` and just returning `true` here and `false` in line 54.

Reply:
The story with `value` is a little more complex than it seems 😁 

The `value` is method's input parameter, and it's passed by `ref bool value` which means reference/pointer, used later internally by the caller (built-in autocomplete Visual Studio API), and not setting `value` explicitly and returning only true/false values could have some side effects. Probably there is a good(?) reason why `ProposalManagerBase.TryGetIsProposalPosition()` signature uses ` ref bool value` and not `bool value` (and your proposition would have perfect sense for this second case).